core/app/models/spree/stock_movement.rb

```diff
@@ -6,10 +6,17 @@ class StockMovement < Spree::Base
     after_create :update_stock_item_quantity
 
     validates :stock_item, presence: true
-    validates :quantity, presence: true
+    validates :quantity, presence: true, numericality: {
+                 greater_than_or_equal_to: -2**31,
+                 less_than_or_equal_to: 2**31-1,
+                 only_integer: true,
+                 allow_nil: true
+              }
```

Comment:
Indent the right brace the same as the start of the line where the left brace is.